### DISCUSSION TITLE
More info in CloseCode.SubprotocolNotAcceptable situations?

### DISCUSSION THREAD

**Discussion by theogravity** [enisdenjo/graphql-ws#339] [Q&A] [answered]
We've been seeing this in our error logs, but we're not sure what the source is. We have converted all the stuff we know of to Apollo 3 / graphql-ws. There could be some chance someone is using something with an older client, but it's hard to tell because our error logs only print this message:

```
WebSocket protocol error occured. It was most likely caused due to an unsupported subprotocol "" requested by the client. graphql-ws implements exclusively the "graphql-transport-ws" subprotocol, please make sure that the client implements it too.
```

Is there a way for us to try / catch somewhere on the server-side so we can add context in? We're using Apollo Express + ws.

**Comment by enisdenjo** [comment, marked as answer]
Some of your clients are trying to connect without specifying a subprotocol, while the required subprotocol is `"graphql-transport-ws"`.

I've added that log so that people transitioning from `subscriptions-transport-ws` are aware that this lib is not cross-compatible. ~~Maybe I should console log exclusively if the deprecated subprotocol is detected?~~ Starting from [v5.6.4](https://github.com/enisdenjo/graphql-ws/releases/tag/v5.6.4), server integrations warn [only when detecting `subscriptions-transport-ws` clients](https://github.com/enisdenjo/graphql-ws/commit/e08073956621b52d858b617eff5cf993ccedea16).

The log comes from [here](https://github.com/enisdenjo/graphql-ws/blob/d54f98fdbc85f330b8febf08ad53ffca23a9dd37/src/use/ws.ts#L160-L168) and can be disabled by running the server in production mode (`export NODE_ENV=production`).

**Reply by theogravity** [reply]
> Some of your clients are trying to connect without specifying a subprotocol, while the required subprotocol is `"graphql-transport-ws"`.
> 
> I've added that log so that people transitioning from `subscriptions-transport-ws` are aware that this lib is not cross-compatible. Maybe I should console log exclusively if the deprecated subprotocol is detected? 🤔
> 
> The log comes from [here](https://github.com/enisdenjo/graphql-ws/blob/db0ccf5d7b631a66c6e49f1145942445bda7cd76/src/use/ws.ts#L161-L166), you can disable it by running the server in production mode (`export NODE_ENV=production`).

I think what's curious is that the subprotocol in the log doesn't specify a subprotocol at all. We were using the old transport lib for our apollo 2 stuff, not sure if that included the `graphql-ws` subprotocol marker.

**Reply by enisdenjo** [reply]
> I think what's curious is that the subprotocol in the log doesn't specify a subprotocol at all.

That is a client-side issue unrelated to graphql-ws.

Furthermore, starting from [v5.6.4](https://github.com/enisdenjo/graphql-ws/releases/tag/v5.6.4), server integrations warn [only when detecting `subscriptions-transport-ws` clients](https://github.com/enisdenjo/graphql-ws/commit/e08073956621b52d858b617eff5cf993ccedea16).
